### PR TITLE
CAR-6339: feat: extended checkboxFilter component with tracking

### PR DESCRIPTION
**Pull request by GutierrezGo**
While working on the checkboxFilter component extraction, we were skipping to pass the tracking. 
Now, the tracking is added as an optional prop.

PR in listings -> https://github.com/carforyou/carforyou-listings-web/pull/3066

## Thank you 🦙

### CODE REVIEW

File: src/components/filters/checkbox.tsx [hunk 1]
```diff
@@ -16,6 +16,7 @@ interface Props {
   apply: (filtersToApply) => void
   selected?: (string | number)[]
   facet?: { [key: string]: number }
+  tracking?: (value) => void
```
Comment (Averethel) [inline src/components/filters/checkbox.tsx:19]: Would it make sense to call it `onSelect` or something like that? In the end, it's a function that you call when a value is selected. It's not really limited to tracking.

File: src/components/filters/checkbox.tsx [hunk 2]
```diff
@@ -24,12 +25,15 @@ const CheckboxFilter: FC<Props> = ({
...
   selected = [],
   facet = {},
+  tracking,
 }) => {
   const values = options.map(({ value }) => value)
 
   const selectValue = (value) => () => {
     const selectionCopy = [...selected]
 
+    tracking ? tracking(value) : null
```
Comment (Averethel) [inline src/components/filters/checkbox.tsx:35]:
```suggestion
    if (tracking) {
      tracking(value)
    }
```

File: src/components/filters/checkbox.tsx (hunk 1 above)
Comment (Averethel) [inline src/components/filters/checkbox.tsx:19]: Can we add a type to the value here?

**Review by Averethel — CHANGES_REQUESTED** [review]
A good starting point. Please check my comment on naming.

What's missing for me is a test and a story in storybook.

File: src/components/filters/checkbox.tsx (hunk 1 above)
Comment (GutierrezGo) [reply src/components/filters/checkbox.tsx:19]: Yes, it makes sense thanks.

Comment (GutierrezGo) [reply src/components/filters/checkbox.tsx:19]: @Averethel  We can but I was thinking now that this function is not only for tracking maybe is not limited to a type.

File: src/stories/filters/checkbox.stories.tsx [hunk 3]
```diff
@@ -101,3 +101,9 @@ WithIcon.args = {
     },
   ],
 }
+
+export const WithOnSelect = Template.bind({})
+WithOnSelect.args = {
+  storyName: "With onSelect",
+  onSelect: action("on Click"),
```
Comment (Averethel) [inline src/stories/filters/checkbox.stories.tsx:108]:
```suggestion
  onSelect: action("on select"),
```

**Review by Averethel — APPROVED** [review]

**Pull request merged**